src/Interpreters/CollectJoinOnKeysVisitor.cpp

```diff
@@ -16,6 +16,23 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+void CollectJoinOnKeysMatcher::Data::setDisjuncts(const ASTFunction & func)
+{
+    const auto * expression_list = func.children.front()->as<ASTExpressionList>();
+    std::vector<const IAST*> v;
```

Comment:
Will be fixed.

Reply:
The excuse is "why bother using ASTPtr since we use raw pointers to identify nodes anyway"?
So, it makes sense to avoid using raw pointers e.g. in TableJoin::addDisjunct, but I have no idea how.